tests/sentry/buffer/test_redis.py

```diff
@@ -248,6 +249,63 @@ def test_process_pending_partitions_none(self, process_pending, process_incr):
         # Make sure we didn't queue up more
         assert len(process_pending.apply_async.mock_calls) == 2
 
+    def test_enqueue(self):
+        PROJECT_ID_BUFFER_LIST_KEY = "project_id_buffer_list"
+        project_id = 1
+        rule_id = 2
+        group_id = 3
+        event_id = 4
+        group2_id = 5
+        event2_id = 6
+
+        project_id2 = 7
+        rule2_id = 8
+        group3_id = 9
+        event3_id = 10
+
+        # store the project ids
+        self.buf.push_to_set(key=PROJECT_ID_BUFFER_LIST_KEY, value=project_id)
+        self.buf.push_to_set(key=PROJECT_ID_BUFFER_LIST_KEY, value=project_id2)
+
+        # store the rules and group per project
+        self.buf.enqueue(
+            model=Project,
+            field={"project_id": project_id},
+            value={f"{rule_id}:{group_id}": event_id},
+        )
```

Comment:
whe should probably create a test case where we have the same project / rule / group id's here, and that we want to update the `event_id` in the store

Reply:
Oh this won't work, I'll have to rewrite it. `hsetnx` won't do anything if the field for that key already exists.